### PR TITLE
Tenancy cleanup

### PR DESCRIPTION
**Pull request by Tijunel**
Basically added tenancy/auth rules to endpoints as needed. Added a bunch of other endpoints for auth and user, did cleanup where I could. Error handling and tenancy if-else mess could probably use a cleanup in the future with some tenant assert functions, we should review next time we see each other.

### CODE REVIEW

**Review by arhamHumayun — DISMISSED** [review]

File: app/models/operator.model.go [hunk 1]
```diff
@@ -5,7 +5,9 @@ import (
 )
 
 type Operator struct {
-	ID 					primitive.ObjectID 	`json:"_id,omitempty" bson:"_id,omitempty"`
-	Name				string							`json:"name,omitempty" bson:"name,omitempty"`
+	ID 							primitive.ObjectID 		`json:"_id,omitempty" bson:"_id,omitempty"`
```
Comment (justinf34) [inline app/models/operator.model.go:8]: Could we format this?

File: app/models/sensor.model.go [hunk 2]
```diff
@@ -10,17 +10,18 @@ type Sensor struct {
...
 	Disabled             *bool              `json:"disabled,omitempty" bson:"disabled,omitempty"`
 	ThingID              primitive.ObjectID `json:"thingId,omitempty" bson:"thingId,omitempty"`
-	UpperCalibration     *float64           `json:"upperCalibration,omitempty" bson:"upperCalibration,omitempty"`
-	LowerCalibration     *float64           `json:"lowerCalibration,omitempty" bson:"lowerCalibration,omitempty"`
-	ConversionMultiplier *float64           `json:"conversionMultiplier,omitempty" bson:"conversionMultiplier,omitempty"`
-	UpperWarning         *float64           `json:"upperWarning,omitempty" bson:"upperWarning,omitempty"`
-	LowerWarning         *float64           `json:"lowerWarning,omitempty" bson:"lowerWarning,omitempty"`
-	UpperDanger          *float64           `json:"upperDanger,omitempty" bson:"upperDanger,omitempty"`
-	LowerDanger          *float64           `json:"lowerDanger,omitempty" bson:"lowerDanger,omitempty"`
+	UpperCalibration     float64           	`json:"upperCalibration,omitempty" bson:"upperCalibration,omitempty"`
```
Comment (justinf34) [inline app/models/sensor.model.go:19]: I think if we don't set this as a pointer then a 0 value would be considered empty and this not be able to input zeros in the database

File: app/models/thing.model.go [hunk 3]
```diff
@@ -8,5 +8,6 @@ type Thing struct {
 	ID        				primitive.ObjectID   `json:"_id,omitempty" bson:"_id,omitempty"`
```
Comment (justinf34) [inline app/models/thing.model.go:8]: Could we format this file?

File: app/models/thing_operator.model.go [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package models
+
+import (
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+type ThingOperator struct {
+	ID						primitive.ObjectID		`json:"_id,omitempty" bson:"_id,omitempty"`
```
Comment (justinf34) [inline app/models/thing_operator.model.go:8]: Could we format this file?

File: utils/errors.go [hunk 5]
```diff
@@ -22,6 +22,7 @@ const (
 	EntityCreationError 	= "entityCreationError"
```
Comment (justinf34) [inline utils/errors.go:22]: Could we format this file?

**Review by justinf34 — CHANGES_REQUESTED** [review]

File: app/models/sensor.model.go (hunk 2 above)
Comment (Tijunel) [reply app/models/sensor.model.go:19]: Good catch.

Comment (Tijunel) [reply app/models/sensor.model.go:19]: Actually no, we don't want to insert the element at all in the database if it does not get posted or put. Found this out the hard way. This way, the fields do not exist in the db if they are not put in.

**Review by justinf34 — APPROVED** [review]

**Pull request merged**